backend/src/email/services/email-transactional.service.ts

```diff
@@ -74,17 +77,108 @@ async function sendMessage({
     recipients: [recipient],
     replyTo,
     attachments: sanitizedAttachments,
+    referenceId: emailMessageTransactionalId.toString(),
   }
   logger.info({
     message: 'Sending transactional email',
     action: 'sendMessage',
   })
-  const messageId = await EmailService.sendEmail(mailToSend)
+  const messageId = await EmailService.sendEmail(mailToSend, {
+    extraSmtpHeaders: { isTransactional: true },
+  })
   if (!messageId) {
     throw new Error('Failed to send transactional email')
   }
 }
 
+type CallbackMetaData = {
+  timestamp: Date
+  bounce?: {
+    bounceType: string
+    bounceSubType: string
+  }
+  complaint?: {
+    complaintFeedbackType: string
+    complaintSubType: string
+  }
+}
+async function handleStatusCallbacks(
+  type: string,
+  id: string,
+  metadata: CallbackMetaData
+): Promise<void> {
+  switch (type) {
```

Comment:
ohh just found `backend/src/email/utils/callback/update-status.ts` and `backend/src/email/utils/callback/query.ts` — wonder if we can re-use the code there

it seems like what they've done there is to run a logger.info before actually making any db changes:
```
const { errorCode, errorSubType, timestamp, id } = opts

  logger.info({
    message: 'Updating email_messages table',
    opts,
    action: 'updateMessageWithError',
  })
```

maybe we can follow that pattern here? (in case sth goes wrong, so we can have logs to trace)

Reply:
separately, maybe we can just use `void` instead of `await`, since we're not actually waiting for the function to return anything

Reply:
> 

Nope, this was intentionally written without reusing that as we're having 2 separate standards of dealing with callbacks at the time of writing

Reply:
OK with keeping the code separate — any reason why we're not logging the update though?

Reply:
> OK with keeping the code separate — any reason why we're not logging the update though?

Cos we're saving it in database? Would be a waste of computing resource and storage